### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by ha-manel**
## 🎯 In this branch I:
- Installed `redux` and `react-redux`.
- Created `actions` and `reducers` for books and categories.
- Combined the reducers and configured the `redux store`.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+// Action Creators
+export function addBook(book) {
+  return { type: ADD_BOOK, payload: book };
+}
+
+export function removeBook(bookID) {
+  return { type: REMOVE_BOOK, payload: bookID };
+}
```
Comment (prolajumokeoni) [inline src/redux/books/books.js:23]: - [ ] Kindly make sure that a new book can be  added or  removed, currently I can't remove or see book I added to the array 👍🏿

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @ha-manel, 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Git flow was correctly used
✔️  Good design
✔️  Good PR title and Description



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (ha-manel) [reply src/redux/books/books.js:23]: @prolajumokeoni, Hi! we're only required to create actions, reducers, and the store in this project. using dispatchers to make adding/removing the books functional will be next project's requirements. Please check the following [link](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md) to confirm.

Comment (prolajumokeoni) [reply src/redux/books/books.js:23]: Hi @ha-manel, if you check the [project description](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md#description), it says "In this step, you will configure your Redux Store and write actions and reducer for adding and removing books. NOTE: editing an existing book and updating the progress is not part of this step. Neither is the styling of the application's components **but adding and removing actions needs to work**", please read the project requirements again 👍🏿

Comment (ha-manel) [reply src/redux/books/books.js:23]: @prolajumokeoni  Hi again! I did add `actions` and `reducers` for adding and removing books. for the adding and removing to work you have to add dispatchers not actions and reducers to the store and that step (dispatching) will be done tomorrow in tomorrow's [project](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components.md#project-requirements) (also see screenshot below, they are from tomorrow's project requirements)
image
image

Comment (nicupop729) [reply src/redux/books/books.js:23]: Hi there,

This PR is now under second opinion review. Please have a bit of patience :)

**Review by nicupop729 — COMMENTED** [review]
Hello @ha-manel 👋,

_This is an additional review as a result of the second opinion request._

I am glad to let you know that you are right. You don't have to implement the full functionality at this stage, but only to create your redux store.
In the next stages, you will dispatch in the components the functions for adding and removing books.
Thanks a lot for providing the right links in the thread, so that my job was easier.

@prolajumokeoni, thanks a lot for the review. If the situation repeats with another PR, please recheck the requirements, as now the curriculum is different since we wore students.

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away :)

Keep rocking 🚀!

**Review by keroloussamy — APPROVED** [review]
Hi @ha-manel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by prolajumokeoni** [comment]
Thank you @nicupop729 for the feedback 🤗, all the best  with your project @ha-manel, glad this issue has been resolved 👍🏿

**Comment by ha-manel** [comment]
Thank you for your time! 😊

**Pull request merged**